js/StandardLibraryIntellisense.ts

```diff
@@ -27,6 +33,15 @@ class Sketch {
     AddWire (wire                : oc. TopoDS_Wire) : Sketch;
     
     LineTo  (nextPoint           : number[]       ) : Sketch;
+    HorizontalLineTo(xNextPoint: number): Sketch;
+    HorizontalLineOfLength(xLength: number): Sketch;
+    VerticalLineTo(yNextPoint: number): Sketch;
+    VerticalLineOfLength(yLength: number): Sketch;
+    AngledLineOf(angle: number, lengthOptions: {ofLength?: number, ofVerticalLength?: number, ofHorizontalLength?: number}): Sketch;
+    AngledLineTo(angle: number, point: {toXPoint?: number, toYPoint?: number}): Sketch;
+    Tag(tagName: string): Sketch;
+    AngledLineToTaggedLineIntersect(angle: number, tagName: string, offset: number): Sketch;
```

Comment:
The use case obviously to find the intersection of any two lines, with an offset available for the reference line since the main usecase I had in mind is parallel lines, so here's where I used it.
![image](https://user-images.githubusercontent.com/29681384/104165435-79f13d00-544d-11eb-941d-41ff90c86f97.png)
I had already made the upper/left angled line and so finding the intersection of the lower horizontal line with a line of same angle but offset so that I could keep the thickness of the "arm" consistent as it kinks was the desire. This could be done a "helper function" what would need surrounding information about the lines as params, but since most of the information is already in the sketch I though tagging that part of the geometry was a way to keep the API expressive. I'm not sure there are other usecases? and not sure if "tag" is the best word "id", "ref" or "reference" might also be better?

Reply:
Another thing that's worth mentioning is the offset polarity matters and it's impacted by the "direction" of the line, so basically the offset goes to the left-hand side of the line if you were travelling from the first point to the second. Left because that would be the direction of a positive angle coming from the line, but other than that the convention is arbitrary, (a negative number is needed for right side).

I thought a convention like this was better than something like a positive number always offsets above the original line since that would still need a convention for perfectly vertical lines, and if anyone has a parametric part that has a line that goes between say 85degrees and 95degrees they'd, they have to add logic to stop the switch that would happen when it crosses the threshold.

Open to debate on this one though.